web/src/components/storage/VolumeForm.jsx

```diff
@@ -0,0 +1,478 @@
+    }
+
+    case "UPDATE_DATA": {
+      return {
+        ...state,
+        formData: {
+          ...state.formData,
+          ...payload
+        }
+      };
+    }
+
+    case "SET_ERRORS": {
+      return { ...state, errors: payload };
+    }
+
+    default: {
+      return state;
+    }
+  }
+};
+
+/**
+ * Form used for adding a new file system from a list of templates
+ * @component
+ *
+ * @note VolumeForm does not provide a submit button. It is the consumer's
+ * responsibility to provide both: the button for triggering the submission by
+ * using the form id and the callback function used to perform the submission
+ * once the form has been validated.
+ *
+ * @param {object} props
+ * @param {string} props.id - Form ID
+ * @param {object[]} props.templates - Volume templates
+ * @param {onSubmitFn} props.onSubmit - Function to use for submitting a new volume
+ *
+ * @callback onSubmitFn
+ * @param {object} volume
+ * @return {void}
+ */
+export default function VolumeForm({ id, volume: currentVolume, templates = [], onSubmit }) {
+  const [state, dispatch] = useReducer(reducer, currentVolume || templates[0], createInitialState);
+
+  const changeVolume = (mountPoint) => {
+    const volume = templates.find(t => t.mountPoint === mountPoint);
+    dispatch({ type: "CHANGE_VOLUME", payload: { volume } });
+  };
+
+  const updateData = (data) => dispatch({ type: "UPDATE_DATA", payload: data });
+
+  const validateVolumeSize = (sizeMethod, volume) => {
+    const errors = {};
+    const { minSize, maxSize } = volume;
+
+    switch (sizeMethod) {
+      case SIZE_METHODS.AUTO:
+        break;
+      case SIZE_METHODS.MANUAL:
+        if (!minSize) {
+          errors.size = "A size value is required";
+        }
+        break;
+      case SIZE_METHODS.RANGE:
+        if (!minSize) {
+          errors.minSize = "Minimum size is required";
+        }
+
+        if (maxSize !== -1 && maxSize <= minSize) {
+          errors.maxSize = "Maximum must be greater than minimum";
+        }
+        break;
+    }
+
+    return errors;
+  };
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    const { volume: originalVolume, formData } = state;
+    const volume = createUpdatedVolume(originalVolume, formData);
+    const errors = validateVolumeSize(formData.sizeMethod, volume);
+
+    dispatch({ type: "SET_ERRORS", payload: errors });
+
+    if (!Object.keys(errors).length) onSubmit(volume);
+  };
+
+  return (
+    <Form id={id} onSubmit={submitForm}>
+      <FormGroup isRequired label="Mount point" fieldId="mountPoint">
+        <MountPointFormSelect
+          id="mountPoint"
+          value={state.formData.mountPoint}
+          onChange={changeVolume}
+          volumes={currentVolume ? [currentVolume] : templates}
+          isDisabled={currentVolume !== undefined}
+        />
+      </FormGroup>
+      <FormGroup isRequired label="File system type" fieldId="fsType">
+        <NumericTextInput
```

Comment:
Why a `NumericTextInput`?

Reply:
For limiting the user input to numbers (including floats) only. See 417f8bf0d10d13b82e47faea960ddfe01c74b108 and https://github.com/openSUSE/agama/pull/590#discussion_r1223162713